src/configure

```diff
@@ -121,6 +124,11 @@ function read_dirname {
   echo $retval
 }
 
+function read_value {
+  local val=`expr "X$1" : '[^=]*=\(.*\)'`;
```

Comment:
What happens if this does not match the pattern string?

Reply:
It matches, definitionally, based on its use within the case statement: `--cuda-arch=*)`

Reply:
You are correct.

Reply:
Can you replace other usages of `expr "X$1" : '[^=]*=\(.*\)'` with this function? Fore example, `--fst-version`

Reply:
That should be doable but we wanted to limit the scope of what we were touching.  Adopting this for other configure options seems like it should fall into a separate patch set after this is accepted.

Reply:
It's calling a function wrapping a specific functionality instead of copy-pasta'ing it to various places.
There is no problem with making this kind of cleanup change.

Reply:
Done.